cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -257,6 +252,39 @@ where
     }
 }
 
+/// Construct a cache if one should be used.
+///
+/// Errors if there's an ingress URI, no signing key, and metadata_only is
+/// false.
+fn determine_cache(
+    metadata_only: bool,
+    ingress_uri: Option<Url>,
+    key_file: Option<PathBuf>,
+) -> Result<Option<NixCopyCache>, PublishError> {
+    if metadata_only {
+        return Ok(None);
+    }
+    match (ingress_uri, key_file) {
+        (Some(ingress_uri), Some(key_file)) => Ok(Some(NixCopyCache {
+            url: ingress_uri,
+            key_file,
+        })),
+        (Some(_), None) => Err(PublishError::Catchall(
+            indoc! { "
+               A signing key is required to upload artifacts.
+
+               You can supply a signing key by either:
+               - Providing a path to a key with the '--signing-private-key' option.
+               - Setting it in the config via 'flox config --set publish.signing-key <path>'
```

Comment:
We should fix it this time 😄 
```suggestion
               - Setting it in the config via 'flox config --set publish.signing-private-key <path>'
```

Reply:
Fixed